tests/queries/0_stateless/03403_makeInterval.sql

```diff
@@ -0,0 +1,30 @@
+SELECT makeInterval('nanosecond', 5) as interval, toDateTime64('2025-01-01 00:00:00', 9) + interval AS res;
+SELECT makeInterval('microsecond', 5) as interval, toDateTime64('2025-01-01 00:00:00', 9) + interval AS res;
+SELECT makeInterval('millisecond', 5) as interval, toDateTime64('2025-01-01 00:00:00', 9) + interval AS res;
+SELECT makeInterval('second', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT makeInterval('Second', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT makeInterval('SECOND', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT makeInterval('Minute', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT makeInterval('Hour', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT makeInterval('Day', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT makeInterval('Week', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT makeInterval('Month', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT makeInterval('Quarter', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT makeInterval('Year', 5) as interval, toDateTime('2025-01-01 00:00:00') + interval AS res;
+SELECT toDateTime('2025-01-01 00:00:00') + makeInterval('Year', 5) AS res;
+SELECT toDateTime('2025-01-01 00:00:00') + makeInterval('second', number) FROM numbers(5);
+SELECT toDateTime('2025-01-01 00:00:00') + makeInterval('second', null);
+SELECT toDateTime('2025-01-01 00:00:01') + makeInterval('second', -1);
+SELECT toDateTime('2025-01-01 00:00:00') + makeInterval('second', 0);
+
+SELECT makeInterval(); -- { serverError 42}
+SELECT makeInterval(''); -- { serverError 42}
+SELECT makeInterval('second'); -- { serverError 42 }
+SELECT makeInterval('second', 5, 10); -- { serverError 42 }
+
+SELECT makeInterval('second', '5'); -- { serverError 43 }
+SELECT makeInterval(10, 5); -- { serverError 43 }
+SELECT makeInterval('second', 1.5); -- { serverError 43 }
```

Comment:
We should be able to use strings/floats and lose precision/cast if needed (see `toIntervalYear` as an example, https://fiddle.clickhouse.com/efa4991c-e5d1-4849-81f9-a890f8443b0f)

Reply:
Done.

In making this change though, I removed the type check on the value argument that was previously in `getReturnTypeImpl`, in favor of allowing the underlying `toInterval<Kind>` function throw an error if the conversion is not possible.

Please let me know if there is another validation you'd like me to add for the input, or if there is a straightforward way to validate in `getReturnTypeImpl` that the `toInterval<Kind>` will be able to do the conversion!

Reply:
I think it should be good